frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-ref.ts

```diff
@@ -0,0 +1,84 @@
+ * If the group will not be present then "core" will be returned as part of the group in reference.
+ * * */
+export const getReference = ({
+  group,
+  version,
+  kind,
+}: K8sGroupVersionKind): K8sResourceKindReference => [group || 'core', version, kind].join('~');
+
+/**
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s model.
+ * @param model k8s model
+ * @return The reference for model i.e `group~version~kind`.
+ * * */
+export const getReferenceForModel = (model: K8sModel): K8sResourceKindReference =>
+  getReference({ group: model.apiGroup, version: model.apiVersion, kind: model.kind });
+
+/**
+ * Provides apiVersion for a k8s model.
+ * @param model k8s model
+ * @return The apiVersion for the model i.e `group/version`.
+ * * */
+export const getAPIVersionForModel: GetAPIVersionForModel = (model) =>
+  !model?.apiGroup ? model.apiVersion : `${model.apiGroup}/${model.apiVersion}`;
+
+/**
+ * Provides a group, version, and kind for a resource.
+ * @param resource k8s resource
+ * @return The group, version, kind for the provided resource.
+ * If the resource does not have an API group, group "core" will be returned.
+ * If the resource has an invalid apiVersion then null will be returned.
+ * * */
+export const getGroupVersionKindForResource: GetGroupVersionKindForResource = (resource) => {
+  const { apiVersion, kind } = resource;
+  const apiVersionSplit = apiVersion.split('/');
+  const apiVersionSplitLen = apiVersionSplit.length;
+  if (apiVersionSplitLen > 2) throw new Error('Provided resource has invalid apiVersion.');
+
+  return {
+    group: apiVersionSplitLen === 2 ? apiVersionSplit[0] : 'core',
+    version: apiVersionSplitLen === 2 ? apiVersionSplit[1] : apiVersion,
+    kind,
+  };
+};
+
+/**
+ * Provides a group, version, and kind for a reference.
+ * @param reference reference for group, version, kind i.e `group~version~kind`.
+ * @return The group, version, kind for the provided reference.
+ * If the group's value is "core" it denotes resource does not have an API group.
+ * * */
+export const getGroupVersionKindForReference = (
+  reference: K8sResourceKindReference,
+): K8sGroupVersionKind => {
+  const referenceSplit = reference.split('~');
+  if (referenceSplit.length > 3) throw new Error('Provided reference is invalid.');
+
+  const [group, version, kind] = referenceSplit;
+  return {
+    group,
+    version,
+    kind,
+  };
+};
+
+/**
+ * Provides a reference string that uniquely identifies the group, version, and kind of K8sGroupVersionKind.
+ * @param kind kind can be of type K8sResourceKindReference or K8sGroupVersionKind
+ * @return The reference i.e `group~version~kind`.
+ * * */
+export const transformGroupVersionKindToReference = (
+  kind: K8sResourceKindReference | K8sGroupVersionKind,
+): K8sResourceKindReference =>
+  kind && typeof kind !== 'string' ? getReference(kind) : (kind as K8sResourceKindReference);
```

Comment:
I'm debating if we need to log a warning saying the reference string is deprecated. My hesitation is that we use reference all over the place in core code, which could create a lot of noise until we fix it. Maybe hold off for now.

Reply:
resolving as no action is needed here.